sentry_streams/src/watermark.rs

```diff
@@ -0,0 +1,144 @@
+use crate::messages::{RoutedValuePayload, WatermarkMessage};
+use crate::routes::{Route, RoutedValue};
+use sentry_arroyo::processing::strategies::{
+    CommitRequest, ProcessingStrategy, StrategyError, SubmitError,
+};
+use sentry_arroyo::types::Message;
+use std::collections::BTreeMap;
+use std::time::{Duration, SystemTime, UNIX_EPOCH};
+
+/// Returns the current Unix epoch
+fn current_epoch() -> u64 {
+    SystemTime::now()
+        .duration_since(UNIX_EPOCH)
+        .unwrap()
+        .as_secs()
+}
+
+pub struct Watermark {
+    pub next_step: Box<dyn ProcessingStrategy<RoutedValue>>,
+    pub route: Route,
+    pub period: u64,
+    last_timestamp: u64,
+}
+
+impl Watermark {
+    /// A strategy that periodically sends watermark messages.
+    /// This strategy is added as the first step in a consumer.
+    /// The Arroyo adapter commit step only commits once it recives a watermark
+    /// message with a specific committable from all branches in a consumer.
```

Comment:
This explains the struct, so it should be put on top of the struct declaration not in the impl.